xfields/beam_elements/beambeam3d.py

```diff
@@ -209,10 +245,21 @@ def __init__(self,
         assert (slices_other_beam_zeta_center is not None
                 or slices_other_beam_zeta_center_star is not None)
         assert slices_other_beam_num_particles is not None
+        assert slices_other_beam_num_macroparticles is not None
```

Comment:
If the slice contains 0 particles, the mean and sigma are 0. If the slice contains 1 particle, the sigma is 0. If the sigma is 0 then the cerrf returns nans, so the field will be nan. Instead of the if on the macroparticles I could add an if on sigma > 0, but there has to be an if anyways.

Reply:
See https://github.com/xsuite/xfields/pull/50#discussion_r1001469988

Reply:
see [4 hours ago](https://github.com/xsuite/xfields/pull/50#discussion_r1003030751)